azure/function-app/storage_account.tf

```diff
@@ -14,6 +14,6 @@ resource "azurerm_storage_account" "managed_storage_account" {
 }
 
 data "azurerm_storage_account" "storage_account" {
-  name                = coalesce(var.storage_account.preexisting_name, azurerm_storage_account.managed_storage_account.0.name)
-  resource_group_name = var.resource_group_name
+  name                = var.storage_account.preexisting_name != null ? var.storage_account.preexisting_name : azurerm_storage_account.managed_storage_account.0.name
```

Comment:
Does `coalesce` not work here?

Reply:
`coalesce` does not short circuit, so evern if `var.storage_account.preexisting_name` is not null, `azurerm_storage_account.managed_storage_account.0.name` is still evaluated, which leads to an error since `count` is 0 if `var.storage_account.preexisting_name` is not null (same reasoning for not using coalesce for `resource_group_name`)